third_party/terraform/resources/resource_dataflow_job.go

```diff
@@ -499,3 +507,36 @@ func resourceDataflowJobIsVirtualUpdate(d *schema.ResourceData) bool {
 
 	return false
 }
+
+func waitForDataflowJobToBeUpdated(d *schema.ResourceData, config *Config, replacementJobID string, timeout time.Duration) error {
+	return resource.Retry(timeout, func() *resource.RetryError {
+		project, err := getProject(d, config)
+		if err != nil {
+			return resource.NonRetryableError(err)
+		}
+
+		region, err := getRegion(d, config)
+		if err != nil {
+			return resource.NonRetryableError(err)
+		}
+
+		replacementJob, err := resourceDataflowJobGetJob(config, project, region, replacementJobID)
+		if err != nil {
+			if isGoogleApiErrorWithCode(err, 404) {
+				return resource.RetryableError(fmt.Errorf("the replacement job with ID %q was not found.", replacementJobID))
```

Comment:
have you seen the job return a 404 while updating? that feels more like a real error than a retryable, but also APIs can behave weirdly

Reply:
I have not. I added it here since the Dataflow API has behaved in other weird ways in the past, but maybe it is safe to not worry about 404 errors in this case.

@c2thorn have you ever seen a NotFound error when updating Dataflow jobs?

Reply:
Nope. I haven't seen any eventual consistency with this resource, or otherwise weird 404's

Reply:
Ok. Fixed.